crates/iota-names-indexer/src/events.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use iota_names::{config::IotaNamesConfig, domain::Domain, registry::NameRecord};
+use iota_types::{base_types::IotaAddress, event::Event};
+use serde::{Deserialize, Serialize};
+
+pub(crate) enum IotaNamesEvent {
+    IotaNamesRegistry(IotaNamesRegistryEvent),
+    IotaNamesReverseRegistry(IotaNamesReverseRegistryEvent),
+    AuctionStarted(AuctionStartedEvent),
+    AuctionBid(AuctionBidEvent),
+    AuctionExtended(AuctionExtendedEvent),
+    AuctionFinalized(AuctionFinalizedEvent),
+}
+
+impl IotaNamesEvent {
+    pub(crate) fn try_from_event(event: &Event, config: &IotaNamesConfig) -> anyhow::Result<Self> {
+        anyhow::ensure!(
+            event.package_id == config.package_address.into(),
+            "Invalid event package: {}",
+            event.package_id
+        );
+        Ok(match event.type_.name.as_str() {
+            "IotaNamesRegistryEvent" => Self::IotaNamesRegistry(bcs::from_bytes(&event.contents)?),
```

Comment:
The bcs decoding doesn't work, it fails with `unexpected end of input`

Reply:
It's a String in the SC but an actual Domain here